[Related #48] Initialize websockets

supunTE opened the pull request:
Related #48 
Initialized web sockets on the client and the server.

supunTE committed:
build(package.json): install socket io

re #48

supunTE committed:
build(package.json): install socket-io client

re #48

supunTE committed:
build(package.json): install cors

supunTE committed:
feat(server): setup socket-io on server

re #48

supunTE committed:
feat(client): setup socket-io client

re #48

dalanad commented on apps/client/src/services/initialize-socket.ts:
```diff
@@ -0,0 +1,11 @@
+import { io, Socket } from 'socket.io-client';
+
+const SOCKET_ENDPOINT =
+  process.env['NX_SOCKET_ENDPOINT'] || 'http://localhost:3000';
```
Why do we use NX as the prefix ?

supunTE replied:
Nx checks for the`NX` prefix to add when building.

![image](https://user-images.githubusercontent.com/33794760/221827305-8e7dc495-82a5-448a-82b2-4331dcf6047a.png)
https://nx.dev/recipes/environment-variables/use-environment-variables-in-react

dalanad commented on package.json:
```diff
@@ -27,6 +28,7 @@
     "@reduxjs/toolkit": "^1.8.6",
     "@rematch/core": "^2.2.0",
     "core-js": "^3.6.5",
+    "cors": "^2.8.5",
```
Is this package a peer dependency?  I don't see it getting used

supunTE replied:
Mentioned in, 
https://github.com/the-ai-team/razor/pull/53/files#r1119848113

supunTE commented on apps/server/src/main.ts:
```diff
@@ -1,11 +1,33 @@
+  process.env.NX_ALLOWED_ORIGINS.split(', ') || 'http://localhost:4200';
+
+const io = new Server(server, {
+  cors: {
```
@dalanad In here I'm using `cors` package to configure the `Access-Control-Allow-Origin`

dalanad replied:
doesn't it automatically get installed with socket.io?

dalanad replied:
@supunTE

supunTE replied:
![image](https://user-images.githubusercontent.com/33794760/223947938-b71a1648-0c65-4ac5-bf2a-f205a7a33ddb.png)
https://socket.io/docs/v3/handling-cors/

supunTE replied:
removed the package d725ceb

supunTE replied:
Yeah, we don't need cors to install manually. socket IO is providing us with that. I misunderstood the issue.
removed the package https://github.com/the-ai-team/razor/commit/d725ceb42b18d42d53f02fb0d09390fcb0cea0ae

supunTE committed:
Merge branch 'main' into feature/48-initialize-websockets

supunTE committed:
refactor: remove unnecessary file

supunTE committed:
Merge branch 'main' into feature/48-initialize-websockets

supunTE committed:
build(ts): update ts config

supunTE committed:
Revert "build(ts): update ts config"

This reverts commit 102ce2304e286b753039e30bba0fcfe94c4d51ae.

supunTE committed:
build(package.json): remove cors package

cors package does not need to install manually. socket io is already providing that.
